### PR TITLE
[SPARK-48936][CONNECT] Makes spark-shell works with Spark connect

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to add the support of `--remote` at `bin/spark-shell` so it can use Spark Connect easily.

At high level:

```
./bin/spark-shell --remote "local[*]"
...
spark.range(10).show()
```

1. Start `ConnectRepl`
2. Start the Spark Connect server locally at `ConnectRepl.scala`
3. Create a Spark Connect client, and expose it to the shell users
4. When users exit, Spark Connect server is also terminated.

```
./bin/spark-shell --remote "sc://localhost"
...
spark.range(10).show()
```

1. Start `ConnectRepl`
2. Create a Spark Connect client, and expose it to the shell users

The approach of classpath handling is as follows:

- Place all Spark Connect REPL related dependencies and jars into `jars/connect-repl`.
- Only when we run Spark Connect shell, we load the dependencies at `jars/connect-repl` (and exclude Spark Classic jars, and jars shaded in Spark Connect client).

In order to allow configurations set in Spark Shell, this PR leverages properties (that already contains Spark configurations), and set it in the Spark Connect client. 

### Why are the changes needed?

`bin/pyspark --remote` also works. In order for end users to try Spark Connect out, should have the consistent way.

### Does this PR introduce _any_ user-facing change?

Yes, now `bin/spark-shell` supports `--remote` option.

### How was this patch tested?

Manually tested as below:

```
./bin/spark-shell --remote "local[*]"
...
spark.range(10).show()
```

```
./sbin/start-connect-server.sh --wait
...
./bin/spark-shell --remote "sc://localhost"
...
spark.range(10).show()
```

```
./bin/spark-shell --remote "local[*]" --conf spark.sql.ansi.enabled=false
...
spark.conf.get("spark.sql.ansi.enabled")
```

```
./bin/spark-shell --remote "local[*]" --conf spark.sql.ansi.enabled=false --conf spark.abcabc=abc --conf spark.sql.debug=true
...
spark.conf.get("spark.sql.debug")
spark.conf.get("spark.sql.ansi.enabled")
```

```
./sbin/start-connect-server.sh --wait
...
./bin/spark-shell --remote "sc://localhost" --conf spark.sql.ansi.enabled=false --conf spark.abcabc=abc --conf spark.sql.debug=true
...
spark.conf.get("spark.sql.debug")
spark.conf.get("spark.sql.ansi.enabled")
```

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala
Comment (HyukjinKwon) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/application/ConnectRepl.scala:43]: Just to match with/without Spark Connect

File: bin/spark-shell
Comment (dongjoon-hyun) [inline bin/spark-shell:37]: This seems to be first `Spark Classic` wording outside `python`.

```
$ git grep 'Spark Classic'
python/pyspark/errors/error-conditions.json:      "Calling property or member '<member>' is not supported in PySpark Classic, please use Spark Connect instead."
python/pyspark/sql/classic/__init__.py:"""Spark Classic specific"""
python/pyspark/sql/column.py:    # Spark Classic Column by default. This is NOT an API, and NOT supposed to
python/pyspark/sql/connect/expressions.py:        #   Column<'CAST(a AS BIGINT)'>     <- Spark Classic
python/pyspark/sql/dataframe.py:    # Spark Classic DataFrame by default. This is NOT an API, and NOT supposed to
python/pyspark/sql/session.py:        In Spark Classic, a temporary view referenced in `spark.sql` is resolved immediately,
python/pyspark/sql/session.py:        In Spark Classic, a temporary view referenced in `spark.table` is resolved immediately,
python/pyspark/sql/tests/connect/test_connect_dataframe_property.py:            # Using this temp env to properly invoke mapInPandas in PySpark Classic.
python/pyspark/sql/tests/connect/test_parity_types.py:    @unittest.skip("This test is dedicated for PySpark Classic.")
python/pyspark/sql/tests/connect/test_parity_types.py:    @unittest.skip("This test is dedicated for PySpark Classic.")
```

**Review by dongjoon-hyun — COMMENTED** [review]
Could you re-trigger the failed CI?

File: bin/spark-shell
Comment (HyukjinKwon) [reply bin/spark-shell:37]: Let me actually just start a vote for the naming...

Comment (dongjoon-hyun) [reply bin/spark-shell:37]: +1 for that. Thank you!

Comment (HyukjinKwon) [reply bin/spark-shell:37]: Sent out! Let me hold off this PR till we set it down

**Comment by HyukjinKwon** [comment]
Code should be ready (but held off for the naming Spark Classic - https://lists.apache.org/thread/ys7zsod8cs9c7qllmf0p0msk6z2mz2ym)

**Comment by HyukjinKwon** [comment]
Vote passed. This should be ready for a look, cc @hvanhovell

**Review by dongjoon-hyun — APPROVED** [review]
Thank you so much, @HyukjinKwon .

**Comment by pan3793** [comment]
thanks for making this change, if I understand correctly, we don't need to manually install [Coursier CLI](https://get-coursier.io/docs/cli-installation) and `spark-connect-repl` after this patch, if so, the docs should be updated too.

image

**Comment by HyukjinKwon** [comment]
Oh yeah. Should update the docs too.

**Comment by HyukjinKwon** [comment]
Will push some changes later.

**Comment by HyukjinKwon** [comment]
Actually for docs, I will make a separate PR. I would need to make few more PRs in any event.

**Comment by HyukjinKwon** [comment]
The build works with SBT but doesn't with Maven - investigating

File: bin/spark-shell
Comment (pan3793) [inline bin/spark-shell:85]: is this a temporary workaround or designed as a long-term solution?

I know that we can not put `spark-connect-client-jvm` into `jars` due to class name issues, but this requires the user to download the jars from the internet (or an private maven repo) when executing `spark-shell --remote xxx` for the first time , this may not be a good solution for users deploying Spark to environments with restricted Internet access.

How about including these jars into spark binary tgz in a different folder?

Comment (HyukjinKwon) [reply bin/spark-shell:85]: It is temporary one.

> How about including these jars into spark binary tgz in a different folder?

Yeah, this was exactly what I was thinking :-).

Comment (HyukjinKwon) [reply bin/spark-shell:85]: I will have to make multiple PRs related to this (e.g., fixing Spark Submit). I will make sure addressing this before Spark 4.0 release.

Comment (HyukjinKwon) [reply bin/spark-shell:85]: Let me actually address it now.

**Comment by HyukjinKwon** [comment]
Alright, I believe I addressed all comments, and this is ready

File: assembly/pom.xml
Comment (pan3793) [inline assembly/pom.xml:179]: this is a little bit hacky, if we use `./dev/make-distribution.sh <extra-args>` to package make a binary distribution, `extra-args` won't be passed into this command, these may cause some issues.

is it possible to collect artifacts in the `spark-connect-client-jvm` module and only perform a simple copy here to avoid recursive `build/mvn` calls?

also cc @LuciferYang, do you have any suggestions?

Comment (HyukjinKwon) [reply assembly/pom.xml:179]: I took a look actually but seems like there aren't other ways around

Comment (pan3793) [reply assembly/pom.xml:179]: Try with `dev/make-distribution.sh -Phive` but failed.
```
[INFO] --- exec:3.2.0:exec (get-connect-client-jar) @ spark-assembly_2.13 ---
cp: /Users/chengpan/Projects/apache-spark/assembly/../connector/connect/client/jvm/target/spark-connect-client-jvm_2.13-4.0.0-SNAPSHOT.jar: No such file or directory
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
```

Comment (HyukjinKwon) [reply assembly/pom.xml:179]: That shouldn't be related to the profiles. Let me take a look.

Comment (pan3793) [reply assembly/pom.xml:179]: anyway, thanks for addressing the packaging issue, i think we can adopt this approach(after you fix the command) as a temporary solution to unblock your other changes, and investigate how to improve it later.

Comment (HyukjinKwon) [reply assembly/pom.xml:179]: Nah thanks for taking a look. It was a real issue so I fixed it 👍

Comment (LuciferYang) [reply assembly/pom.xml:179]: When I execute `build/mvn package -DskipTests -pl assembly -am`, only 24 modules are built. Since `assembly` does not depend on the `connect-client-jvm` module, the `connect-client-jvm` module is not built, leading to a similar error mentioned by @pan3793  earlier. Maybe this is the same issue?

```
[INFO] --- exec:3.2.0:exec (get-connect-client-jar) @ spark-assembly_2.13 ---
cp: /Users/yangjie01/SourceCode/git/spark-mine-13/assembly/../connector/connect/client/jvm/target/spark-connect-client-jvm_2.13-4.0.0-SNAPSHOT.jar: No such file or directory
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:355)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:253)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:884)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:844)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:450)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:903)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:280)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:203)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Spark Project Parent POM 4.0.0-SNAPSHOT:
[INFO] 
[INFO] Spark Project Parent POM ........................... SUCCESS [  2.381 s]
[INFO] Spark Project Tags ................................. SUCCESS [  4.217 s]
[INFO] Spark Project Sketch ............................... SUCCESS [  6.052 s]
[INFO] Spark Project Common Utils ......................... SUCCESS [ 15.559 s]
[INFO] Spark Project Local DB ............................. SUCCESS [  5.974 s]
[INFO] Spark Project Networking ........................... SUCCESS [  7.780 s]
[INFO] Spark Project Shuffle Streaming Service ............ SUCCESS [  6.530 s]
[INFO] Spark Project Variant .............................. SUCCESS [  2.353 s]
[INFO] Spark Project Unsafe ............................... SUCCESS [  7.575 s]
[INFO] Spark Project Launcher ............................. SUCCESS [  4.775 s]
[INFO] Spark Project Core ................................. SUCCESS [01:08 min]
[INFO] Spark Project ML Local Library ..................... SUCCESS [ 14.716 s]
[INFO] Spark Project GraphX ............................... SUCCESS [ 16.256 s]
[INFO] Spark Project Streaming ............................ SUCCESS [ 23.458 s]
[INFO] Spark Project SQL API .............................. SUCCESS [ 12.449 s]
[INFO] Spark Project Catalyst ............................. SUCCESS [ 58.710 s]
[INFO] Spark Project SQL .................................. SUCCESS [01:39 min]
[INFO] Spark Project ML Library ........................... SUCCESS [01:05 min]
[INFO] Spark Project Connect Common ....................... SUCCESS [ 21.806 s]
[INFO] Spark Avro ......................................... SUCCESS [ 18.595 s]
[INFO] Spark Protobuf ..................................... SUCCESS [ 18.957 s]
[INFO] Spark Project REPL ................................. SUCCESS [ 11.450 s]
[INFO] Spark Project Connect Server ....................... SUCCESS [ 29.381 s]
[INFO] Spark Project Assembly ............................. FAILURE [ 27.147 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE

```

Comment (HyukjinKwon) [reply assembly/pom.xml:179]: Yeah I'll push a fix

File: pom.xml
Comment (LuciferYang) [inline pom.xml:333]:
```
[WARNING] Some problems were encountered while building the effective model for org.apache.spark:spark-sketch_2.13:jar:4.0.0-SNAPSHOT
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.apache.spark:spark-kvstore_2.13:jar:4.0.0-SNAPSHOT
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.apache.spark:spark-network-common_2.13:jar:4.0.0-SNAPSHOT
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.apache.spark:spark-network-shuffle_2.13:jar:4.0.0-SNAPSHOT
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.apache.spark:spark-unsafe_2.13:jar:4.0.0-SNAPSHOT
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
```
Should use ` ${project.version}` ?

File: assembly/pom.xml
Comment (LuciferYang) [inline assembly/pom.xml:191]: If we change the scope of `ammonite` in `connect-client-jvm` to compile, do we still need to explicitly download these dependencies? I'm not sure if this will bring any negative impacts?

Comment (HyukjinKwon) [reply assembly/pom.xml:191]: `connect-client-jvm` is not the part of the default packaging so `ammonite` wouldn't be available. Another option is shading but it's too much .. ammonite is too large and unstable in my opinion, and we should actually get rid of this when possible.

**Comment by HyukjinKwon** [comment]
tests should pass now.

**Comment by HyukjinKwon** [comment]
Made a PR for doc: https://github.com/apache/spark/pull/47676

**Comment by LuciferYang** [comment]
```
-rw-r--r--  1 yangjie01  staff     62783  4 30 17:53 acyclic_2.13.14-0.3.12.jar
-rw-r--r--  1 yangjie01  staff     37463  5 14 04:13 ammonite-compiler-interface_2.13.14-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff    406291  5 14 04:13 ammonite-compiler_2.13.14-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff     28079  5 14 04:14 ammonite-interp-api_2.13.14-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff    234191  5 14 02:12 ammonite-interp_2.13.14-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff     83540  5 14 02:12 ammonite-repl-api_2.13.14-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff    105093  5 14 02:13 ammonite-repl_2.13.14-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff    193030  5 14 02:13 ammonite-runtime_2.13.14-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff    178177  5 14 02:15 ammonite-terminal_2.13-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff    154569  5 14 02:15 ammonite-util_2.13-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff    177407  5 14 02:16 ammonite_2.13.14-3.0.0-M2.jar
-rw-r--r--  1 yangjie01  staff      3185  5 14 02:16 ammonite_2.13.14-3.0.0-M2.pom
-rw-r--r--  1 yangjie01  staff    151956  9 24  2023 bsp4j-2.1.0-M5.jar
-rw-r--r--  1 yangjie01  staff    214381  2  7  2023 checker-qual-3.5.0.jar
-rw-r--r--  1 yangjie01  staff      8572  5 12  2023 class-path-util_2.13-0.1.4.jar
-rw-r--r--  1 yangjie01  staff   2487631  3 29 16:04 common_2.13-4.9.3.jar
-rw-r--r--  1 yangjie01  staff     10673  6 12  2023 dependency-interface_2.13-0.2.3.jar
-rw-r--r--  1 yangjie01  staff     92642  6 12  2023 dependency_2.13-0.2.3.jar
-rw-r--r--  1 yangjie01  staff     13879  2  7  2023 error_prone_annotations-2.3.4.jar
-rw-r--r--  1 yangjie01  staff      4617  2  7  2023 failureaccess-1.0.1.jar
-rw-r--r--  1 yangjie01  staff     65360  4 15 10:15 fansi_2.13-0.5.0.jar
-rw-r--r--  1 yangjie01  staff    331335  4 15 12:15 fastparse_2.13-3.1.0.jar
-rw-r--r--  1 yangjie01  staff     85575  4 14 22:17 geny_2.13-1.1.0.jar
-rw-r--r--  1 yangjie01  staff    283367  4 20  2023 gson-2.10.1.jar
-rw-r--r--  1 yangjie01  staff   2862361  8  9  2023 guava-30.1-jre.jar
-rw-r--r--  1 yangjie01  staff   9943108  9 25  2023 interface-1.0.19.jar
-rw-r--r--  1 yangjie01  staff      8781  2  7  2023 j2objc-annotations-1.3.jar
-rw-r--r--  1 yangjie01  staff     72305  3 16  2023 java-diff-utils-4.12.jar
-rw-r--r--  1 yangjie01  staff    882046  7 29  2019 javaparser-core-3.2.12.jar
-rw-r--r--  1 yangjie01  staff    734815  2  8  2023 javassist-3.21.0-GA.jar
-rw-r--r--  1 yangjie01  staff   1423747  1 23  2024 jline-3.25.1.jar
-rw-r--r--  1 yangjie01  staff    161361  3 25  2023 jline-reader-3.14.1.jar
-rw-r--r--  1 yangjie01  staff    214856  3 25  2023 jline-terminal-3.14.1.jar
-rw-r--r--  1 yangjie01  staff     62969  3 25  2023 jline-terminal-jna-3.14.1.jar
-rw-r--r--  1 yangjie01  staff   1505196  3 25  2023 jna-5.3.1.jar
-rw-r--r--  1 yangjie01  staff     19936  2  7  2023 jsr305-3.0.2.jar
-rw-r--r--  1 yangjie01  staff     35755  1 21  2024 lenses_2.13-0.11.15.jar
-rw-r--r--  1 yangjie01  staff      2199  2  7  2023 listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
-rw-r--r--  1 yangjie01  staff    199201  9 18  2023 mainargs_2.13-0.5.4.jar
-rw-r--r--  1 yangjie01  staff     54029  9 24  2023 org.eclipse.lsp4j.generator-0.20.1.jar
-rw-r--r--  1 yangjie01  staff    137420  9 24  2023 org.eclipse.lsp4j.jsonrpc-0.20.1.jar
-rw-r--r--  1 yangjie01  staff    112545  9 24  2023 org.eclipse.xtend.lib-2.28.0.jar
-rw-r--r--  1 yangjie01  staff     79739  9 24  2023 org.eclipse.xtend.lib.macro-2.28.0.jar
-rw-r--r--  1 yangjie01  staff    166255  9 24  2023 org.eclipse.xtext.xbase.lib-2.28.0.jar
-rw-r--r--  1 yangjie01  staff    367244  4 15 10:15 os-lib_2.13-0.10.0.jar
-rw-r--r--  1 yangjie01  staff    127127  4 15 10:15 pprint_2.13-0.9.0.jar
-rw-r--r--  1 yangjie01  staff   1664824  5 12  2023 protobuf-java-3.19.6.jar
-rw-r--r--  1 yangjie01  staff    124948  4  5 13:14 requests_2.13-0.8.2.jar
-rw-r--r--  1 yangjie01  staff      7091  4 16 03:58 scala-collection-compat_2.13-2.12.0.jar
-rw-r--r--  1 yangjie01  staff  12245472  4 30 08:07 scala-compiler-2.13.14.jar
-rw-r--r--  1 yangjie01  staff   5924141  4 30 08:07 scala-library-2.13.14.jar
-rw-r--r--  1 yangjie01  staff   3745916  7 17  2023 scala-reflect-2.13.10.jar
-rw-r--r--  1 yangjie01  staff    478051  7 21  2021 scala-xml_2.13-2.0.1.jar
-rw-r--r--  1 yangjie01  staff    389798  4 15 12:15 scalaparse_2.13-3.1.0.jar
-rw-r--r--  1 yangjie01  staff   2419295  1 21  2024 scalapb-runtime_2.13-0.11.15.jar
-rw-r--r--  1 yangjie01  staff     41125  2  7  2023 slf4j-api-1.7.36.jar
-rw-r--r--  1 yangjie01  staff    119360  4 16 08:19 sourcecode_2.13-0.4.1.jar
-rw-r--r--  1 yangjie01  staff  23498566  8  9 10:30 spark-connect-client-jvm_2.13-4.0.0-SNAPSHOT.jar
-rw-r--r--  1 yangjie01  staff      9701  4  1  2023 typename_2.13-1.1.0.jar
-rw-r--r--  1 yangjie01  staff    238836  9 16  2023 ujson_2.13-3.1.3.jar
-rw-r--r--  1 yangjie01  staff    109674  9 16  2023 upack_2.13-3.1.3.jar
-rw-r--r--  1 yangjie01  staff    204735  9 16  2023 upickle-core_2.13-3.1.3.jar
-rw-r--r--  1 yangjie01  staff    243701  9 16  2023 upickle-implicits_2.13-3.1.3.jar
-rw-r--r--  1 yangjie01  staff    114688  9 16  2023 upickle_2.13-3.1.3.jar
```

There are 64 jar files collected in the `connect-repl` directory, do we really need all of them? Some of them, such as `acyclic_2.13.14-0.3.12.jar`, are not present in the dependency tree of the `connect-client-jvm` module. Additionally, Spark currently uses slf4j 2.x, but the `connect-repl` directory contains the slf4j-api 1.7.5 version. Won't this cause version conflicts?

**Comment by HyukjinKwon** [comment]
Those dependencies were pulled by Maven's `dependency:copy-dependencies` for `ammonite` alone so it should be correct.

In case of SBT, it brings more unnecessary dependencies but should be fine because we don't use SBT for the official release.

Even some version conflicts, `bin/spark-shell --remote ...` is new, and old code won't be affected.

**Review by LuciferYang — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]
+1 for merging it as-is, and improving the package later.

**Comment by HyukjinKwon** [comment]
Merged to master.

File: bin/spark-shell
Comment (HyukjinKwon) [inline bin/spark-shell:47]: Made a PR to remove this ugly hack: https://github.com/apache/spark/pull/47702

File: assembly/pom.xml
Comment (HyukjinKwon) [inline assembly/pom.xml:180]: Made a PR to prune unnecessary jars (https://github.com/apache/spark/pull/47705), still testing.

**Pull request closed without merging**
